### ISSUE TITLE
Create an OpenShift project that holds the different elements inside

### ISSUE THREAD

**Issue by chargio** [project-koku/koku#15]
All the different elements of the infrastructure should be available inside a project in OpenShift (possibly the Insights one is the best platform)
- [ ] SSO (transient, until Insights provides its own one)
- [ ] Data gathering / cleaning (transient, until Insights provides the final one)
- [ ] Database layer for configuration
- [ ] Historical data layer (transient, until Insights provides the final implementation)
- [ ] Cost calculation API service (to be able to calculate savings and best options)
- [ ] Reporting UI
- [ ] Core
- [ ] CI/CD, inc. dev --> test --> preproduction --> production workflows

**Comment by chargio** [comment]
The different elements can be under the same pod / infrastructure, once we understand that the objective is to keep them under different microservices in the future.
There are some decisions to be taken then:
- Everything should be available through an API if we are going to split it into microservices. It can be done now or in the future.
- That API can be available under different protocols. REST, gRPC, GraphQL. At least REST should be an option, but it seems that gRPC is better suited for internal comms.
- Our architecture does not need to support this in the first instalment, but it should be ready to be moved to that once we start needing it

**Comment by blentz** [comment]
This is a huge epic. We'll need to break it down before we have actionable tasks.

**Comment by chargio** [comment]
Yes. I expect this to be split in actionable items during our sprint preparation meetings, and then be acted upon.

**Comment by chambridge** [comment]
Closing this issue as we have accomplished many components for OSD deployment with our API layer and database. We will be relying on our API layer for user registry and token validation. We are deploying via Github webhook into insights. Remaining pieces like UI and backend data processing will have there own stories for deployment.

UI component has the following issue: https://github.com/project-koku/koku-ui/issues/11
